soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -289,6 +290,172 @@ void Randomizer::LoadHintLocations(const char* spoilerFileName) {
+    RC_MARKET_POTION_SHOP_ITEM_2,
+    RC_MARKET_POTION_SHOP_ITEM_3,
+    RC_MARKET_POTION_SHOP_ITEM_4,
+    RC_MARKET_POTION_SHOP_ITEM_5,
+    RC_MARKET_POTION_SHOP_ITEM_6,
+    RC_MARKET_POTION_SHOP_ITEM_7,
+    RC_MARKET_POTION_SHOP_ITEM_8,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_1,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_2,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_3,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_4,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_5,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_6,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_7,
+    RC_MARKET_BOMBCHU_SHOP_ITEM_8,
+};
+
+// Reference soh/src/overlays/actors/ovl_En_GirlA/z_en_girla.h
+std::unordered_map<RandomizerGet, int32_t> randomizerGetToEnGirlShopItem = {
+    { RG_BUY_DEKU_NUT_5,    0x00 },
+    { RG_BUY_ARROWS_30,     0x01 },
+    { RG_BUY_ARROWS_50,     0x02 },
+    { RG_BUY_BOMBS_525,     0x03 },
+    { RG_BUY_DEKU_NUT_10,   0x04 },
+    { RG_BUY_DEKU_STICK_1,  0x05 },
+    { RG_BUY_BOMBS_10,      0x06 },
+    { RG_BUY_FISH,          0x07 },
+    { RG_BUY_RED_POTION_30, 0x08 },
+    { RG_BUY_GREEN_POTION,  0x09 },
+    { RG_BUY_BLUE_POTION,   0x0A },
+    { RG_BUY_HYLIAN_SHIELD, 0x0C },
+    { RG_BUY_DEKU_SHIELD,   0x0D },
+    { RG_BUY_GORON_TUNIC,   0x0E },
+    { RG_BUY_ZORA_TUNIC,    0x0F },
+    { RG_BUY_HEART,         0x10 },
+    { RG_BUY_BOMBCHU_10,    0x15 },
+    { RG_BUY_BOMBCHU_20,    0x16 },
+    { RG_BUY_DEKU_SEEDS_30, 0x1D },
+    { RG_BUY_BLUE_FIRE,     0x27 },
+    { RG_BUY_BOTTLE_BUG,    0x28 },
+    { RG_BUY_POE,           0x2A },
+    { RG_BUY_FAIRYS_SPIRIT, 0x2B },
+    { RG_BUY_ARROWS_10,     0x2C },
+    { RG_BUY_BOMBS_20,      0x2D },
+    { RG_BUY_BOMBS_30,      0x2E },
+    { RG_BUY_BOMBS_535,     0x2F },
+    { RG_BUY_RED_POTION_40, 0x30 },
+    { RG_BUY_RED_POTION_50, 0x31 },
+};
+
+void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
+    if (strcmp(spoilerFileName, "") != 0) {
+        ParseHintLocationsFile(spoilerFileName);
+    }
+
+    CustomMessageManager::Instance->ClearMessageTable(Randomizer::merchantMessageTableID);
+    CustomMessageManager::Instance->AddCustomMessageTable(Randomizer::merchantMessageTableID);
+    CustomMessageManager::Instance->CreateMessage(Randomizer::merchantMessageTableID, 0,
+        { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+            "\x12\x38\x82\All right! You win! In return for&sparing me, I will give you a&%gmysterious item%w!&Please, take it!\x07\x10\xA3",
+            "\x12\x38\x82\In Ordnung! Du gewinnst! Im Austausch&dafür, dass du mich verschont hast,&werde ich dir einen %gmysteriösen&Gegenstand%w geben! Bitte nimm ihn!\x07\x10\xA3",
+            "\x12\x38\x82\D'accord! Vous avez gagné! En échange&de m'épargner, je vous donnerai un &%gobjet mystérieux%w! S'il vous plaît,&prenez-le!\x07\x10\xA3",
+        });
+    
+    // Currently these are generated at runtime, one for each price between 0-95. We're soon going to migrate this
+    // to being generated at save load, with only messages specific to each scrub.
```

Comment:
Is there an issue for actually cleaning up the generation of these messages in the future? Just so it's not forgotten.

Or potentially, we could only have 2 messages in total, one for the "free" variant and one where the rupee amount is inserted when the textbox is requested ingame (just like in the random rupee names). I'm not sure which of the 2 is better though.

Reply:
#1493